src/category_theory/abelian/transfer.lean

```diff
@@ -189,4 +197,167 @@ def abelian_of_equivalence
+  {A X Y : 𝓐} (g : X ⟶ RJ_of A) (f : X ⟶ Y) [mono f] :
+  L.obj Y ⟶ injective.under (L.obj A) :=
+let factors := (injective.injective_under $ L.obj A).factors in
+(factors ((adj.hom_equiv X $ injective.under $ L.obj A).symm g) (L.map f)).some
+
+lemma L_map_comp_to_J_eq [preserves_finite_limits L]
+  {A X Y : 𝓐} (g : X ⟶ RJ_of A) (f : X ⟶ Y) [mono f] :
+  L.map f ≫ (LY_to_J _ adj g f) = (adj.hom_equiv X $ injective.under _).symm g :=
+let factors := (injective.injective_under $ L.obj A).factors in
+(factors ((adj.hom_equiv _ _).symm g) (L.map f)).some_spec
+
+
+/--
+If `g : X → R(J)` and `f : X → Y` is mono in `𝓐`, then there is an morphism `Y → R(J)`
+See the diagram below:
+
+𝓐                                                  𝓑
+
+A ---> R(J) <---                                   L(A) -----> J <--------
+      /        |                                              /          |
+     /         |                                             /           |
+    /  g   by adjunction                                    /            |
+   /           |                                           /             |
+  /            |                                          /        by injectivity
+X              |                                      L(X)               |
+|              |                                       |                 |
+v              |                                       v                 |
+Y --------------                                      L(Y) ---------------
+
+-/
+def Y_to_RJ [preserves_finite_limits L]
+  {A X Y : 𝓐} (g : X ⟶ RJ_of A) (f : X ⟶ Y) [mono f] : Y ⟶ RJ_of A :=
+adj.hom_equiv _ _ $ LY_to_J _ adj g f
+
+lemma comp_Y_to_RJ [preserves_finite_limits L]
+  {A X Y : 𝓐} (g : X ⟶ RJ_of A) (f : X ⟶ Y) [mono f] :
+  f ≫ Y_to_RJ _ adj g f = g :=
+begin
+  have := L_map_comp_to_J_eq _ adj g f,
+  rw ←adj.hom_equiv_apply_eq at this,
+  rw [←this],
+  simp only [LY_to_J, Y_to_RJ, L_map_comp_to_J_eq, hom_equiv_counit, hom_equiv_counit,
+    functor.map_comp, category.assoc, counit_naturality, left_triangle_components_assoc,
+    hom_equiv_naturality_left, hom_equiv_unit],
+  generalize_proofs h₁ h₂,
+  rw [←R.map_comp],
+  simp only [counit_naturality_assoc, left_triangle_components_assoc, functor.map_comp,
+    unit_naturality_assoc],
+  congr,
+  ext,
+  rw h₁.some_spec,
+end
+
+include adj
+
+lemma injective_RJ [preserves_finite_limits L] (A : 𝓐) :
+  injective (RJ_of A) :=
+⟨λ X Y g f m, ⟨by { resetI, exact Y_to_RJ _ adj g f }, by apply comp_Y_to_RJ⟩⟩
+
+/-- the morphism `A → R(J)` obtained by `L(A) → J` via adjunction, this morphism is mono, so that
+`A → R(J)` is an injective presentation of `A` in `𝓐`.-/
+def to_RJ (A : 𝓐) :
+  A ⟶ RJ_of A :=
+adj.hom_equiv A (injective.under $ L.obj A) (injective.ι _)
+
+local notation `to_RJ_of` A := to_RJ adj A
+
+instance mono_to_RJ (A : 𝓐)
+  [abelian 𝓐] [abelian 𝓑] [preserves_finite_colimits L] [preserves_finite_limits L] [faithful L] :
```

Comment:
`preserves_finite_colimits L`  already follows from the fact that `L` is a left adjoint, right?